packages/terra-tabs/src/_CollapsibleTabs.jsx

```diff
@@ -165,60 +163,25 @@ class CollapsibleTabs extends React.Component {
 
     // We don't want menu keydown events to propagate and conflict when the tabs keydown events
     // Instead of stopping menu key event propagation, we whitelist event.targets so we do tab focus mgmt only on tab based event targets
-    const tabList = event.target.getAttribute('role') === 'tablist';
-    const tabMoreBtn = event.target.getAttribute('data-terra-tabs-menu') === 'true';
-
-    if (tabList || tabMoreBtn) {
-      const isRTL = document.getElementsByTagName('html')[0].getAttribute('dir') === 'rtl';
-      const visibleChildren = this.container.children;
-
-      if (event.nativeEvent.keyCode === KeyCode.KEY_LEFT) {
-        if (isRTL) {
-          this.handleFocusRight(visibleChildren, event);
-        } else {
-          this.handleFocusLeft(visibleChildren, event);
-        }
-      } else if (event.nativeEvent.keyCode === KeyCode.KEY_RIGHT) {
-        if (isRTL) {
-          this.handleFocusLeft(visibleChildren, event);
-        } else {
-          this.handleFocusRight(visibleChildren, event);
-        }
+    const tabList = event.target.getAttribute('role') === 'tab';
+    const isRTL = document.getElementsByTagName('html')[0].getAttribute('dir') === 'rtl';
+    const nextKey = !isRTL ? KeyCode.KEY_RIGHT : KeyCode.KEY_LEFT;
+    const previousKey = !isRTL ? KeyCode.KEY_LEFT : KeyCode.KEY_RIGHT;
+    let tabPanes = this.container.querySelectorAll('[role="tab"]');
+    tabPanes = (Array.from(tabPanes).filter((pane) => pane.hasAttribute('data-terra-tabs-show-focus-styles') && pane.getAttribute('data-terra-tabs-show-focus-styles') === 'true'));
+    const index = tabPanes.indexOf(event.target);
+    if (tabList) {
+      if ((event.nativeEvent.keyCode === KeyCode.KEY_RETURN || event.nativeEvent.keyCode === KeyCode.KEY_SPACE) && !event.target.getAttribute('data-terra-tabs-menu')) {
```

Comment:
when we press space page gets scrolled to down. you may need event.preventDefault() here to prevent page from scrolling

https://user-images.githubusercontent.com/12869809/230483601-cc975fcf-b23d-4703-8520-dff7ab3bc6b5.mov